src/models/costdelegate.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+  costdelegate.cpp
+
+  This file is part of Hotspot, the Qt GUI for performance analysis.
+
+  Copyright (C) 2016-2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+  Author: Milian Wolff <milian.wolff@kdab.com>
+
+  Licensees holding valid commercial KDAB Hotspot licenses may use this file in
+  accordance with Hotspot Commercial License Agreement provided with the Software.
+
+  Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+  This program is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 2 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "costdelegate.h"
+
+#include "treemodel.h"
+
+#include <QDebug>
+#include <QPainter>
+
+#include <cmath>
+
+CostDelegate::CostDelegate(QObject* parent)
+    : QStyledItemDelegate(parent)
+{
+}
+
+CostDelegate::~CostDelegate() = default;
+
+void CostDelegate::paint(QPainter* painter, const QStyleOptionViewItem& option, const QModelIndex& index) const
+{
+    // TODO: handle negative values
+    const int64_t cost = index.data(AbstractTreeModel::SortRole).toULongLong();
+    if (cost == 0) {
+        QStyledItemDelegate::paint(painter, option, index);
+        return;
+    }
+
+    const int64_t totalCost = index.data(AbstractTreeModel::TotalCostRole).toULongLong();
+    // top-down can miscalculate the peak cost
```

Comment:
I think this might actually not be required anymore - could you check whether removing the std::min(1, ) wrapper? the abs and division is still fine though.